<?xml version="1.0"?>

<launch>
  <node name="espros_nogui" pkg="espros_nogui" type="espros_nogui">

    <!--  Used with rosrun in console.
          Supports use of single output with out this .launch file.
          If set, overrides the below params.

          0 - distance
          1 - distance color
          2 - grayscale
          3 - interleaved distance and amplitude
          4 - amplitude
          -1 - default, ignores this param, allowing use below params -->
    <param name="espros_data" value="-1" />

    <param name="show_distance" value="1" />
    <param name="show_distance_color" value="1" /> <!-- Primarily for human visualization -->
    <param name="show_grayscale" value="0" /> <!-- Trumps all others; must be requested from device alone -->
    <param name="show_interleave" value="1" /> <!-- distance and amplitude -->
    <param name="show_amplitude" value="1" />
    <param name="confidence_bits" value="0" /> <!-- does not apply to distance_color -->

    <param name="orient_vertical" value="0" /> <!-- reverse image vertically -->
    <param name="orient_horizontal" value="0" /> <!-- reverse image horizontally -->


    <!-- integration time max values inferred from Espros GUI UI -->
    <!-- 1 - 2,700 (us), 0 to disable, default = 100 -->
    <param name="integration_time_0" value="100" />

    <!-- 1 - 2,500 (us), 0 to disable default 1500-->
    <param name="integration_time_1" value="1500" />

    <!-- 1 - ? (us), 0 to disable -->
    <param name="integration_time_2" value="0" />

    <!-- 1 - 50,000 (us), 0 to disable, default = 15000 -->
    <param name="integration_time_grayscale" value="15000" />

    <!-- 1 - 15,000 (mm), 0 to disable -->
    <param name="offset" value="0" />

    <!-- 1 - 6,000 (lsb), 0 to disable, default = 75 -->
    <param name="min_amplitude" value="75" />

    <!-- Doesn't seem to have any impact on data -->
    <!-- 1 - 15,000 (mm), 0 to disable, default = 4000 -->
    <param name="range" value="4000" />

  </node>
</launch>
